src/tests/unit_asio_stream.cpp

```diff
@@ -0,0 +1,772 @@
+/*
+* TLS ASIO Stream Unit Tests
+* (C) 2018-2019 Jack Lloyd
+*     2018-2019 Hannes Rantzsch, Tim Oesterreich, Rene Meusel
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_TLS) && defined(BOTAN_HAS_BOOST_ASIO)
+
+// first boost version to include boost/beast/experimental/test/stream.hpp
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106800
```

Comment:
As I understand it asio_stream.h itself is ok starting with Boost 1.66. In that case can we here include asio_stream.h if Boost >= 1.66 but then only compile the tests if Boost >= 1.68. Which at least allows us to check that the header parses ok with the current compiler.

Reply:
That makes sense. I'll just move the check for 1.68 below the include, as the stream itself checks for 1.66 already.